### PR TITLE
Add NT_SIGINFO NOTE to ELF dumps

### PR DESCRIPTION
**Pull request by mikem8361**
Linux Watson needs this to better triage ELF dumps.

Add CreateDumpOptions helper struct to pass all the command options around. Add the "--code", "--errno", "--address" command line options used to fill the NT_SIGINFO NOTE. The runtime passes to createdump on a crash.

Added "ExceptionType" field to "Parameters" section of the Linux crash report json.

### CODE REVIEW

**Comment by mikem8361** [comment]
Issue: https://github.com/dotnet/runtime/issues/40958

**Comment by mikem8361** [comment]
@hoyosjs ping

File: src/coreclr/pal/src/thread/process.cpp [hunk 1]
```diff
@@ -2009,6 +2009,28 @@ PROCFormatInt(ULONG32 value)
     return buffer;
 }
 
+/*++
+Function:
+    PROCFormatInt64
+
+    Helper function to format an ULONG32 as a string.
```
Comment (hoyosjs) [inline src/coreclr/pal/src/thread/process.cpp:2016]: Doc comment should be ULONG64

**Review by hoyosjs — APPROVED** [review]
I don't think this is quite complete, although I am not sure if we care. System dumps also have the signal info in PRSTATUS under pr_info and potentially pr_cursig, although the last one is non-trivial for us to implement. Also, we are putting the process group in the PID field of the note. These two don't block this PR.

File: src/coreclr/debug/createdump/main.cpp [hunk 2]
```diff
@@ -183,7 +203,7 @@ int __cdecl main(const int argc, const char* argv[])
             else
             {
 #ifdef HOST_UNIX
-                pid = atoi(*argv);
+                options.Pid = atoi(*argv);
```
[the remark this answers is not in the document]
Comment (hoyosjs) [reply src/coreclr/debug/createdump/main.cpp:206]: Opened https://github.com/dotnet/runtime/issues/82989 for this.

[the remark this answers is not in the document]
Comment (mikem8361) [reply src/coreclr/debug/createdump/main.cpp:206]: Yes, createdump, etc. only handles one exception/signal.  Suspending the process should serialize when multiple threads have exceptions to just the first one, but probably multiple threads are trying to exec createdump before the process is suspended by the first one.

**Pull request merged**